src/platform/backends/shared/base_virtual_machine.cpp

```diff
@@ -526,8 +526,6 @@ void BaseVirtualMachine::restore_snapshot(const std::string& name, VMSpecs& spec
     assert_vm_stopped(state); // precondition
 
     auto snapshot = get_snapshot(name);
-
-    // TODO@snapshots convert into runtime_error (persisted info could have been tampered with)
```

Comment:
What was the reason/resolution for removing this?

Reply:
I implemented it in the snapshot: https://github.com/canonical/multipass/pull/3265/commits/d007d26db7098f8ebc086af1ada275b0f34532f9

Better fail early in this case and refuse to load the snapshot at all than attempt to run with faulty contents.

Reply:
Got it! Thanks for the explanation